tests/test_query.py

```diff
@@ -215,7 +220,10 @@ def test_check_user_resource_paths_prefixes(
     assert res.status_code == 201, res.text
 
     # check whether the resource path matches the request we created
-    data = {"resource_paths": [resource_path_to_match]}
+    data = {
+        "resource_paths": [resource_path_to_match],
+        "permissions": test_data["permissions"],
```

Comment:
before this change we were only testing the case of permissions not being provided, and now we're only testing the case of permissions being provided, and only when the permissions match.

unit tests should only test one functionality, so i think you could keep the existing tests as they were (removing `permissions` from `test_data`), and create new tests. Test cases off the top of my head: permissions not provided and matching (maybe unnecessary if already covered), permissions not provided and not matching, permissions provided and matching, permissions provided and not matching (partial mismatch), permissions provided and not matching (complete mismatch)

Maybe the expanded permissions could be provided to `list_policies_patcher` instead of just the IDs, if that helps